pkg/types/grammar_types.go

```diff
@@ -434,6 +434,44 @@ const (
 type TableOfContentsMacro struct {
 }
 
+// ------------------------------------------
+// Custom Macro
+// ------------------------------------------
+
+const (
+	// InlineMacro a inline user macro
+	InlineMacro UserMacroKind = "inline"
```

Comment:
Good idea to use a `Kind` field to distinguish between inline and block macros. Would you mind you renaming this `UserMacroKind` type to just `MacroKind`, please, so I can use it to refactor the `InlineImage` and `BlockImage` types and merge them into a single `Image` type?

Reply:
I will rename `UserMacroKind` with `MacroKind`, but I believe should be refactoring `InlineImage` and `BlockImage` in other PR.

Reply:
yes, I'll take care of this in another PR, but having a generic name for the kind will make it easier, which is why I asked you if you could take care about it in your PR ;)